stm32loader.cpp

```diff
@@ -1,161 +1,173 @@
-	return STM32_COMM_ACK;
+char getACKByte()
+{
+    return STM32_COMM_ACK;
 }
 
-char getNACKByte() {
-	return STM32_COMM_NACK;
+char getNACKByte()
+{
+    return STM32_COMM_NACK;
 }
 
-bool isAcknowdledge(int res) {
-	bool ret = false;
-	if (res == getACKByte())
-		ret = true;
-	return ret;
+bool isAcknowdledge(int res)
+{
+    bool ret = false;
+    if (res == getACKByte()) {
+        ret = true;
+    }
+    return ret;
 }
 
-
-int stm32loader::BootLoader::sendCommand(Communication::Commands Command, bool sendInverted) {
-	char commandArray[] = {(char)Command, (char)(~Command)};
-	if (sendInverted)
-		return comPort->sendData(commandArray, 2);
-	else
-		return comPort->sendData(commandArray, 1);
+int stm32loader::BootLoader::sendCommand(Communication::Commands Command, bool sendInverted)
+{
+    char commandArray[] = {(char)Command, (char)(~Command)};
+    if (sendInverted) {
+        return comPort->sendData(commandArray, 2);
+    } else {
+        return comPort->sendData(commandArray, 1);
+    }
 }
 
-int stm32loader::BootLoader::stm32_init() {
-	char readBuffer[comPort->getBuffSize()] = { '\0' };
-	size_t len = comPort->getBuffSize();
+int stm32loader::BootLoader::stm32_init()
+{
+    char readBuffer[comPort->getBuffSize()] = { '\0' };
+    size_t len = comPort->getBuffSize();
 
-	sendCommand(Communication::STM32_CMD_INIT, false);
-	comPort->receiveData(readBuffer, &len);
+    sendCommand(Communication::STM32_CMD_INIT, false);
+    comPort->receiveData(readBuffer, &len);
 
-	int res = readBuffer[0];
+    int res = readBuffer[0];
 
-	if (!isAcknowdledge(res)) {
-		return Communication::STM32_COMM_ERROR;
-	}
+    if (!isAcknowdledge(res)) {
+        return Communication::STM32_COMM_ERROR;
+    }
 
-	return STM32_COMM_ACK == res || res == STM32_COMM_NACK ?
-			Communication::STM32_OK : Communication::STM32_INIT_ERROR;
+    return STM32_COMM_ACK == res || res == STM32_COMM_NACK ?
+           Communication::STM32_OK : Communication::STM32_INIT_ERROR;
 }
 
-int stm32loader::BootLoader::stm32_erase_flash(void) {
-	size_t buffsize = comPort->getBuffSize();
-	char readBuffer[buffsize] = { '\0' };
-	char tries = 2;
-	bool result = false;
-
-
-	sendCommand(Communication::STM32_CMD_ERASE_FLASH);
-	comPort->receiveData(readBuffer, &buffsize);
-
-	if(isAcknowdledge(readBuffer[0]))
-	{
-		readBuffer[0] = '\0';
-		for(int i = 0; i < tries; i++) {
-			sendCommand(Communication::STM32_PARAM_COMPLETE_ERASE);
-			comPort->receiveData(readBuffer, &buffsize);
-
-			if (isAcknowdledge(readBuffer[0])) {
-				i = tries;
-				result = true;
-				break;
-			}
-		}
-	}
-
-	return result == true ? Communication::STM32_OK : Communication::STM32_COMM_ERROR;
+int stm32loader::BootLoader::stm32_erase_flash(void)
+{
+    size_t buffsize = comPort->getBuffSize();
+    char readBuffer[buffsize] = { '\0' };
+    char tries = 2;
+    bool result = false;
+
+    sendCommand(Communication::STM32_CMD_ERASE_FLASH);
+    comPort->receiveData(readBuffer, &buffsize);
```

Comment:
So, jetzt hab ich dich! Du bist in dein eigenes Fettnaepfchen getappt. Denn du nun den Rueckgabewert nicht auswertest, erhaeltst du eine endlos große Buffersize zurueck Beispiel: die Readfunktion liefert -1, und du wandelst das dann in eine unsigned Zahl um, dadurch erkennst du den Fehler nicht und erhaeltst die groeßte darstellbare zahl im Size_t bereich